app/models/vm/operations.rb

```diff
@@ -14,7 +14,11 @@ module Vm::Operations
   end
 
   def cockpit_url
-    URI::HTTP.build(:host => ipaddresses.first, :port => 9090).to_s
+    if MiqServer.active_miq_servers.where(:has_active_cockpit_ws => true)
```

Comment:
I still don't understand this @petervo.  Why is there a `has_active_cockpit_ws` flag instead of looking for a server with the role active?